ophyd/areadetector/plugins.py

```diff
@@ -632,6 +633,56 @@ class ROIPlugin(PluginBase):
                   doc='Minimum size of the ROI in XYZ',
                   default_read_attrs=('min_x', 'min_y', 'min_z'))
 
+    def set(self, region):
+        ''' This functions allows for the ROI regions to be set.
+
+        This function takes in an ROI_number, and a dictionary of tuples and sets the ROI region.
+
+        PARAMETERS
+        ----------
+        region: dictionary.
+            A dictionary defining the region to be set, which has the structure:
+            { 'x':[min, size], 'y':[min, size], 'z':[min, size]}. 
+            Any of the keywords can be ommitted, and they will be ignored.
+        '''
+        if region is not None:
+            for direction, value in region.items():
+                getattr(self, f'min_xyz.min_{direction}').put(value[0])
+                getattr(self, f'size.{direction}').put(value[1])
+
+        return NullStatus()
+        
+    def read(self):
+        '''This function reads back the current ROI status and size.
+
+        This function returns the ROI status and size for each of the ROI's for the detector.
+
+        '''
+        out_dict = OrderedDict({})
+        out_dict[f'{self.name}_status'] = {'timestamp': ttime.time(), 'value': self.enable.get() }
+        for direction in ['x', 'y', 'z']:
+            out_dict[f'{self.name}_{direction}'] = {'timestamp': ttime.time(), 
+                                        'value': (getattr(self, f'min_xyz.min_{direction}').get(),
+                                        getattr(self, f'size_{direction}').get())}
+
+        return out_dict
+
+    def dscribe(self):
```

Comment:
`dscribe`

Reply:
Thanks @klauer I got that one in the recent commit where I changed the f string syntax